Command \printindex already defined

Hello,
i'm really would like to use this thesis template, but i get an error while trying to compile the sources.
I have no idea what to do, even tried to remove \printindex occurences, but doesn't work neither.
Here's the full output of my error : 

$ make
NOTE: You may ignore warnings about the following files:
     thesis.d
Makefile:2556: thesis.d: No such file or directory
= thesis.tex --> thesis.d thesis.pdf.1st.make (0-1) =
= thesis.glo --> thesis.gls =
= thesis.idx --> thesis.ind =
= thesis.tex --> thesis.d thesis.pdf.1st.make (1-1) =
= thesis.bib thesis.aux --> thesis.bbl =
/usr/share/texlive/texmf-dist/tex/latex/glossaries/base/glossaries.sty:1413: LaTeX Error: Command \printindex already defined.
               Or name \end... illegal, see p.192 of the manual.
make: *** [thesis.pdf] Error 1

It gives the same \printindex issue when using makepdf or makeps.

Thanks for help,

eoutin

--- a/README.md
+++ b/README.md
@@ -135,7 +135,7 @@ It also supports some custom options.
 
     _Note_: `glossaries` is the package used to create the glossary.
 
-*   `index`: build the index, which you can put at the and of the thesis with
+*   `withindex`: build the index, which you can put at the and of the thesis with
      the following command (it will create a new unnumbered chapter):
 
         \printthesisindex

--- a/cam-thesis.cls
+++ b/cam-thesis.cls
@@ -76,7 +76,7 @@
 % index - puts the index at the end of the thesis.
 %
 \newif\ifcam@index\cam@indexfalse
-\DeclareOption{index}{\cam@indextrue}
+\DeclareOption{withindex}{\cam@indextrue}
 
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

--- a/thesis.tex
+++ b/thesis.tex
@@ -15,13 +15,13 @@
 % Available documentclass options:
 %
 %   <all `report` document class options, e.g.: `a5paper`>
-%   index       - enables the index. New index entries can be added through `\index{my entry}`
+%   withindex   - enables the index. New index entries can be added through `\index{my entry}`
 %   glossary    - enables the glossary.
 %   techreport  - typesets the thesis in the technical report format.
 %   times       - uses the `Times` font.
 %
 % For more info see `README.md`
-\documentclass[index,glossary]{cam-thesis}
+\documentclass[withindex,glossary]{cam-thesis}
 
 
 

